### PR TITLE
Concurrent update run loop (refactor)

### PR DESCRIPTION
**Pull request by MarinPostma**
This PR allows multiple request to the update store to be performed concurently (i.e, one can list updates while an updates in being written to the update store).

### CODE REVIEW

File: meilisearch-http/src/index_controller/update_actor/actor.rs [hunk 1]
```diff
@@ -16,7 +18,7 @@ use super::{PayloadData, Result, UpdateError, UpdateMsg, UpdateStoreStore};
 pub struct UpdateActor<D, S, I> {
     path: PathBuf,
     store: S,
-    inbox: mpsc::Receiver<UpdateMsg<D>>,
+    inbox: Option<mpsc::Receiver<UpdateMsg<D>>>,
```
Comment (irevoire) [inline meilisearch-http/src/index_controller/update_actor/actor.rs:21]: Why do we need to store the inbox as an option?

**Comment by MarinPostma** [comment]
this will be merged after #158

**Comment by MarinPostma** [comment]
bors try

**Review by irevoire — DISMISSED** [review]

**Review by irevoire — APPROVED** [review]

**Review by irevoire — APPROVED** [review]

**Comment by MarinPostma** [comment]
bors merge

**Pull request merged**